lightningd/dual_open_control.c

```diff
@@ -679,7 +679,9 @@ wallet_commit_channel(struct lightningd *ld,
 		return NULL;
 	}
 
-	channel_info->fee_states = new_fee_states(uc, opener, &feerate);
+	channel_info->fee_states = take(new_fee_states(uc,
+						       opener,
+						       &feerate));
```

Comment:
That is almost certainly wrong.  Were you thinking `tal_steal`?  In which case, change `uc` to `channel_info`.

Reply:
we call `new_channel` below, which if `taken`, will properly tal_steal it onto the newly created `channel` instance. updated to make this clearer, and set the context to `NULL`.